python/pyiceberg/expressions/base.py

```diff
@@ -204,6 +210,7 @@ def __eq__(self, other) -> bool:
         )
 
 
+# TODO: this should still exist as a parent, but term should be abstract
```

Comment:
The `UnaryPredicate` and `BoundUnaryPredicate` classes introduced in this PR should implement the `UnboundPredicate` and `BoundPredicate` interfaces, but I didn't want to refactor the entire set of expressions in this PR so I added this TODO as a reminder.